snapshots/overlay/overlay.go

```diff
@@ -67,13 +68,27 @@ func WithUpperdirLabel(config *SnapshotterConfig) error {
 	return nil
 }
 
+// WithMetacopyCtrl enables metacopy option management by containerd
+func WithMetacopyCtrl(config *SnapshotterConfig) error {
+	config.metacopyOn = new(bool)
+	return nil
+}
+
+// WithMetacopy enables metacopy
+// option for overlayfs mount points
+func WithMetacopy(config *SnapshotterConfig) error {
+	*config.metacopyOn = true
+	return nil
+}
+
 type snapshotter struct {
 	root          string
 	ms            *storage.MetaStore
 	asyncRemove   bool
 	upperdirLabel bool
 	indexOff      bool
 	userxattr     bool // whether to enable "userxattr" mount option
+	metacopyOn    *bool
```

Comment:
why can't this be a bool? In other words, is there any difference between `metacopyOn == nil` and `*metacopyOn == false`?

Reply:
There are three options:

1. `metacopy=on` by config.
2. `metacopy=off` by config.
3. metacopy option wasn't specified.

In previous version it was:

```go
type snapshotter struct {
	root          string
	ms            *storage.MetaStore
	asyncRemove   bool
	upperdirLabel bool
	indexOff      bool
	userxattr     bool // whether to enable "userxattr" mount option
	metacopyOn    bool
        metacopyCtrl  bool
}
```

if metacopy option specified -- `metacopyCtrl = true` if it's not -- `metacopyCtrl = false`. We decided to make `metacopyOn` a simple `*bool` only for the sake of simplicity, so now if `metacopyOn = nil` that means a user didn't specify metacopy option.